### DISCUSSION TITLE
Manually change the text color of the progress bar

### DISCUSSION THREAD

**Discussion by louyuoops** [Textualize/rich#1049] [Q&A]
I am writing a top monitor under windows 10 platform,
and for now, 
I am doing somthing for customrize theme of monitor,
but I tried some methods to change progress bar text color,
mainly 2 places text color, 1st is task description, 2nd is percentage text，
as u know, it doesn't work,
so I hope to get some help here;

here is code about progress:
```python
     cpu_cores_load = Progress(
            "{task.description}",
            SpinnerColumn(),
            BarColumn(style=self.__theme["cpu_module"]["bar_background_color"], complete_style=self.__theme["cpu_module"]["bar_progress_color"]),
            TextColumn("[progress.percentage]{task.percentage:>3.0f}%", style="bright_yellow"),
        )
     for i in sorted(cpu_info["load"]):
            if i == 0:
                name = "cpu"
            else:
                name = "core" + str(i)
            task = cpu_cores_load.add_task(name, total=100)
            cpu_cores_load.update(task, advance = cpu_info["load"][i])
```

here is the top monitor probably looks like (the orange square positions are target text that I wanna change color)
(of course, there are still many features left)
![p2](https://user-images.githubusercontent.com/27338833/108736102-41747100-756c-11eb-9965-cbab43236a23.png)

**Comment by willmcgugan** [comment]
You can use console markup in the text or TextColumns

e.g. to change the description:

```
"[bright_green]{task.description}"
```

```
TextColumn("[progress.percentage]{task.percentage:>3.0f}%", style="bright_yellow"),
```

Here you are setting the style "bright_yellow", but it is being overriden by "[progress.percentage]" which sets a theme style called `progress.percentage`.

**Comment by mezhgano** [comment]
> Here you are setting the style "bright_yellow", but it is being overriden

Thanks for the explanation, but how do we set a custom color in such case?

I tried to change style like this:

`TaskProgressColumn(style='red italic')`

Funny, it is changed to italic, but no to red or any other color...

UPDATE: ok, i got it, you just use the `TextColumn` and remove `[progress.percentage]` style tag, like this:

`TextColumn("{task.percentage:>3.0f}%", style="bright_yellow")`

Then color is changed.
